07_errors/patrickmarabeas/store_test.go

```diff
@@ -14,70 +14,99 @@ type StoreSuite struct {
 	a.Equal(id, 1)
+	a.Equal(error, nil)
+}
+
+func (suite *StoreSuite) TestCreateNegativeNumber() {
+	a := assert.New(suite.T())
+
+	id, error := suite.store.Create(Puppy{Breed: "Wolf", Color: "Grey", Value: -100})
+	a.Equal(id, -1)
+	a.Equal(error, NewError(NegativeValue))
 }
 
 func (suite *StoreSuite) TestRead() {
 	a := assert.New(suite.T())
-	data := suite.store.Read(0)
 
-	a.Equal(data, Puppy{ID: 0, Breed: "Wolf", Color: "Grey", Value: "450"})
+	data, error := suite.store.Read(0)
+	a.Equal(data, Puppy{ID: 0, Breed: "Wolf", Color: "Grey", Value: 450})
+	a.Equal(error, nil)
 }
 
 func (suite *StoreSuite) TestReadNonExistent() {
 	a := assert.New(suite.T())
-	success := suite.store.Read(100)
 
+	success, error := suite.store.Read(100)
 	a.Equal(success, Puppy{})
+	a.Equal(error, NewError(IDNotFound))
 }
 
 func (suite *StoreSuite) TestUpdate() {
 	a := assert.New(suite.T())
-	success := suite.store.Update(0, Puppy{Breed: "Doberman", Color: "Black", Value: "500"})
-	data := suite.store.Read(0)
 
+	success, error := suite.store.Update(0, Puppy{Breed: "Doberman", Color: "Black", Value: 500})
 	a.Equal(success, true)
-	a.Equal(data, Puppy{ID: 0, Breed: "Doberman", Color: "Black", Value: "500"})
+	a.Equal(error, nil)
+
+	data, error := suite.store.Read(0)
+	a.Equal(data, Puppy{ID: 0, Breed: "Doberman", Color: "Black", Value: 500})
+	a.Equal(error, nil)
 }
 
 func (suite *StoreSuite) TestUpdateNonExistent() {
 	a := assert.New(suite.T())
-	success := suite.store.Update(100, Puppy{Breed: "Doberman", Color: "Black", Value: "500"})
 
+	success, error := suite.store.Update(100, Puppy{Breed: "Doberman", Color: "Black", Value: 500})
+	a.Equal(success, false)
+	a.Equal(error, NewError(IDNotFound))
+}
+
+func (suite *StoreSuite) TestUpdateNegativeNumber() {
+	a := assert.New(suite.T())
+
+	success, error := suite.store.Update(0, Puppy{Breed: "Doberman", Color: "Black", Value: -500})
 	a.Equal(success, false)
+	a.Equal(error, NewError(NegativeValue))
 }
 
 func (suite *StoreSuite) TestDestroy() {
 	a := assert.New(suite.T())
-	success := suite.store.Destroy(1)
-	data := suite.store.Read(1)
 
+	success, error := suite.store.Destroy(1)
 	a.Equal(success, true)
+	a.Equal(error, nil)
+
+	data, error := suite.store.Read(1)
 	a.Equal(data, Puppy{})
+	a.Equal(error, NewError(IDNotFound))
 }
 
 func (suite *StoreSuite) TestDestroyNonExistent() {
 	a := assert.New(suite.T())
-	success := suite.store.Destroy(100)
 
+	success, error := suite.store.Destroy(100)
 	a.Equal(success, false)
+	a.Equal(error, NewError(IDNotFound))
 }
 
 func (suite *StoreSuite) TestIdIncrementOnDelete() {
 	a := assert.New(suite.T())
-	id := suite.store.Create(Puppy{Breed: "Greyhound", Color: "Light Brown", Value: "700"})
-	suite.store.Destroy(id)
-	newID := suite.store.Create(Puppy{Breed: "Greyhound", Color: "Light Brown", Value: "700"})
+	id, _ := suite.store.Create(Puppy{Breed: "Greyhound", Color: "Light Brown", Value: 700})
```

Comment:
Rather than ignoring the error here, perhaps use `require`?
```
import "github.com/stretchr/testify/require"
...
require.NoError(suite.T(), err)
```

`require` will stop the test there, and I find it useful to use it to differentiate between test setup that is failing and the results of the tests themselves (for which I use assert).